### PR TITLE
[CI] E2E task refactor

### PR DESCRIPTION
**Pull request by aelovikov-intel**
* Remove temporary experimental task sycl_linux_gen12_exp.yml
* Add possibility to trigger E2E testing on a nightly image manually So far it needs a branch inside intel/llvm but I plan on extending it to PRs later.

### CODE REVIEW

File: .github/workflows/linux_single_e2e.yml [hunk 1]
```diff
@@ -59,6 +59,43 @@ on:
...
+        options:
+          - 'ghcr.io/intel/llvm/sycl_ubuntu2204_nightly:latest'
+          - 'ghcr.io/intel/llvm/sycl_ubuntu2204_nightly:build'
+      image_options:
+        description: |
+          Use option with "--device=/dev/kfd" for AMDGPU, without it for the rest.
+        type: choice
+        options:
+          - '-u 1001 --device=/dev/dri --privileged --cap-add SYS_ADMIN'
+          - '-u 1001 --device=/dev/dri --device=/dev/kfd --privileged --cap-add SYS_ADMIN'
```
Comment (steffenlarsen) [inline .github/workflows/linux_single_e2e.yml:82]: Why do we want to add `SYS_ADMIN` privs to the docker? Do we already do that?

**Review by steffenlarsen — APPROVED** [review]
Let's give it a go!

File: .github/workflows/linux_single_e2e.yml (hunk 1 above)
Comment (bader) [reply .github/workflows/linux_single_e2e.yml:82]: > Why do we want to add `SYS_ADMIN` privs to the docker?

This is still unanswered.
According to oneAPI documentation, this capability is needed for Intel Adviser/Inspector/Vtune tools (https://www.intel.com/content/www/us/en/docs/oneapi-base-toolkit/get-started-guide-linux/2023-2/using-containers-with-the-command-line.html).
AFAIK, our CI doesn't use these tools. Am I missing something?

Comment (aelovikov-intel) [reply .github/workflows/linux_single_e2e.yml:82]: I think we need it for the GPU reset on Intel GPU HW and I wanted to keep options close between these two options here. We might want to lessen those later but I'm focusing on the CI scripts at the moment.

Comment (bader) [reply .github/workflows/linux_single_e2e.yml:82]: Thanks.

> I think we need it for the GPU reset on Intel GPU HW

Should we put this as a comment in the code?

Comment (aelovikov-intel) [reply .github/workflows/linux_single_e2e.yml:82]: Once we spend time to investigate and reduce the options we should document ones that will remain :)

**Pull request merged**